### PR TITLE
Add basic validation to login forms

### PR DESCRIPTION
**Pull request by bp323**

### CODE REVIEW

**Comment by bp323** [comment]
Assigning to @Coenego for review.

**Comment by bp323** [comment]
As a plus, this PR also fixes the issue with the state being lost after logging in.

**Comment by bp323** [comment]
The tests will only start succeeding once #267 goes in

File: shared/gh/css/gh.base.css [hunk 1]
```diff
@@ -70,6 +70,33 @@ select[disabled],
...
+
+.has-error .form-control::-moz-placeholder {
+    font-weight: 400;
+}
+
+.has-error .form-control:-ms-input-placeholder {
+    font-weight: 400;
+}
+
+.has-error .form-control::-webkit-input-placeholder {
```
Comment (Coenego) [inline shared/gh/css/gh.base.css]: Maybe we can do the following?:

```
.has-error .form-control::-moz-placeholder,
.has-error .form-control:-ms-input-placeholder,
.has-error .form-control::-webkit-input-placeholder {
    font-weight: 400;
}
```

File: shared/gh/css/gh.skin.css [hunk 2]
```diff
@@ -57,6 +57,33 @@ a,
...
+
+.has-error .form-control::-moz-placeholder {
+    color: #A94442;
+}
+
+.has-error .form-control:-ms-input-placeholder {
+    color: #A94442;
+}
+
+.has-error .form-control::-webkit-input-placeholder {
```
Comment (Coenego) [inline shared/gh/css/gh.skin.css]: Same here, we should combine the placeholder selectors

File: apps/admin/ui/js/index.js [hunk 3]
```diff
@@ -99,22 +104,21 @@ define(['gh.core', 'gh.constants', 'chosen'], function(gh, constants) {
...
-                var state = $.param(History.getState().data);
-                if (state) {
+        // Log in to the system if the form is valid
+        if (!ev.isDefaultPrevented()) {
+            // Collect and submit the form data
+            var formValues = _.object(_.map($(this).serializeArray(), _.values));
+            gh.api.authenticationAPI.login(formValues.username, formValues.password, function(err) {
+                if (!err) {
                     return window.location.reload();
+                } else {
```
Comment (Coenego) [inline apps/admin/ui/js/index.js]: I guess we can drop the `} else {` here and maybe just do:

```
if (err) {
    return gh.utils.notification('Could not sign you in', 'Please check that you are entering a correct username & password', 'error');
}
window.location.reload();

File: apps/timetable/admin/ui/js/index.js [hunk 4]
```diff
@@ -216,17 +221,13 @@ define(['gh.core', 'gh.constants', 'gh.listview', 'gh.admin-listview', 'gh.admin
...
-        if (gh.config.enableLocalAuth) {
-
+        // Log in to the system if the form is valid
+        if (!ev.isDefaultPrevented() && gh.config.enableLocalAuth) {
             // Collect and submit the form data
             var formValues = _.object(_.map($(this).serializeArray(), _.values));
             gh.api.authenticationAPI.login(formValues.username, formValues.password, function(err) {
                 if (!err) {
-                    window.location = '/admin';
+                    return window.location.reload();
```
Comment (Coenego) [inline apps/timetable/admin/ui/js/index.js]: Same here, we can probably drop the `} else {`. WDYT?

**Comment by Coenego** [comment]
Back to @bp323 with some comments and questions.

File: shared/gh/css/gh.skin.css (hunk 2 above)
Comment (bp323) [inline shared/gh/css/gh.skin.css]: Placeholder styles can't be combined as the styles won't be applied properly in the browser.

File: shared/gh/css/gh.base.css (hunk 1 above)
Comment (bp323) [inline shared/gh/css/gh.base.css]: This won't work, if the rules for the placeholder aren't individually declared browsers will skip over them.

**Comment by bp323** [comment]
Followed up, back to @Coenego

**Pull request merged**